Firebase/Auth/Source/AuthProvider/OAuth/FIROAuthCredential.m

```diff
@@ -43,6 +44,12 @@ - (instancetype)initWithProviderID:(NSString *)providerID
                        accessToken:(nullable NSString *)accessToken
                             secret:(nullable NSString *)secret
                       pendingToken:(nullable NSString *)pendingToken {
+  if ([providerID isEqual:FIRFacebookAuthProviderID]) {
```

Comment:
This check should be under the initializer of OAuthProvider instead of credential.

Reply:
oops. my bad. I was in oauth provider but I clicked the wrong method in IDE and it jumped to the wrong file.